### PR TITLE
stubsabot PR messages: link to diff between releases on GitHub, where possible

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

File: scripts/stubsabot.py [hunk 1]
```diff
@@ -183,6 +183,70 @@ def get_updated_version_spec(spec: str, version: packaging.version.Version) -> s
     return _check_spec(".".join(rounded_version) + ".*", version)
 
 
+@functools.cache
+def get_github_api_headers() -> dict[str, str]:
```
Comment (hauntsaninja) [inline scripts/stubsabot.py:187]: nit: if we're caching, should type as Mapping

File: scripts/stubsabot.py [hunk 2]
```diff
@@ -183,6 +183,70 @@ def get_updated_version_spec(spec: str, version: packaging.version.Version) -> s
...
+    """
+    github_info = await get_github_repo_info(session, pypi_info)
+    if github_info is None:
+        return None
+    tags_to_versions = {tag["name"]: packaging.version.Version(tag["name"]) for tag in github_info.tags}
+    curr_release = packaging.version.Version(stub_info.version_spec.removesuffix(".*")).release
+
+    try:
+        new_tag = next(tag for tag, version in tags_to_versions.items() if version == pypi_info.version)
+        old_tag = next(tag for tag, version in tags_to_versions.items() if version.release[: len(curr_release)] == curr_release)
```
Comment (hauntsaninja) [inline scripts/stubsabot.py:239]: maybe slightly more accurate as the equivalent of:
old_version = max(v for v in versions if v in packaging.specifiers.SpecifierSet(f"=={stub_info.version_spec}"))

**Review by hauntsaninja — COMMENTED** [review]
Thanks, this is useful!

File: scripts/stubsabot.py (hunk 1 above)
Comment (AlexWaygood) [reply scripts/stubsabot.py:187]: Ah, good point!

**Review by hauntsaninja — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks @hauntsaninja! :D

**Pull request merged**